storage/shard.go

```diff
@@ -1533,10 +1533,6 @@ func (s *dbShard) FetchBlocksMetadataV2(
 					blockStart, err)
 			}
 
-			// Make sure ID and tags get cleaned up after read is done
-			ctx.RegisterFinalizer(id)
```

Comment:
hm is this because the id/tags are set to NonFinalizable anyway?

Reply:
This is because the series ID and Tags can both be safely finalized as they are non-finalizeable, so we just finalize when we close the MetadataResult now rather than using the context array.